packages/unchained-client/src/utxo/parser/thorchain.ts

```diff
@@ -0,0 +1,249 @@
+import type { Axios } from 'axios'
+import axios from 'axios'
+
+import { type BaseTxMetadata, Dex, TradeType } from '../../types'
+import type { SubParser, Tx, TxSpecific } from '../parser'
+
+export type LiquidityType = 'Savers' | 'LP'
+export type SwapType = 'Standard' | 'Streaming'
+
+interface Liquidity {
+  type: LiquidityType
+}
+
+interface Swap {
+  type: SwapType
+}
+
+export interface TxMetadata extends BaseTxMetadata {
+  parser: 'thorchain'
+  memo: string
+  liquidity?: Liquidity
+  swap?: Swap
+}
+
+interface Coin {
+  amount: string
+  asset: string
+}
+
+interface InOut {
+  txID: string
+  address: string
+  coins: Coin[]
+  height?: string
+}
+
+interface Metadata {
+  addLiquidity?: {
+    liquidityUnits: string
+  }
+  refund?: {
+    affiliateAddress: string
+    affiliateFee: string
+    memo: string
+    networkFees: Coin[]
+    reason: string
+  }
+  swap?: {
+    affiliateAddress: string
+    affiliateFee: string
+    isStreamingSwap: boolean
+    liquidityFee: string
+    networkFees: Coin[]
+    swapSlip: string
+    swapTarget: string
+    memo: string
+  }
+  withdraw?: {
+    asymmetry: string
+    basisPoints: string
+    impermanentLossProtection: string
+    liquidityUnits: string
+    memo: string
+    networkFees: Coin[]
+  }
+}
+
+interface Actions {
+  date: string
+  height: string
+  in: InOut[]
+  out: InOut[]
+  metadata?: Metadata
+  pools: string[]
+  status: 'success' | 'pending'
+  type: 'swap' | 'addLiquidity' | 'withdraw' | 'donate' | 'refund' | 'switch'
+}
+
+export interface ActionsResponse {
+  actions: Actions[]
+  count: string
+  meta: {
+    nextPageToken: string
+    prevPageToken: string
+  }
+}
+
+interface ExtraMetadata {
+  method: string
+  liquidity?: Liquidity
+  swap?: Swap
+}
+
+const getLiquidityType = (pool: string): LiquidityType => (pool.includes('/') ? 'Savers' : 'LP')
+
+const getSwapType = (memo: string): SwapType => {
+  const regex = /:\d+\/\d+\/\d+:/
+  return regex.test(memo) ? 'Streaming' : 'Standard'
+}
+
+export interface ParserArgs {
+  midgardUrl: string
+}
+
+export class Parser implements SubParser<Tx> {
+  private readonly axiosMidgard: Axios
+
+  constructor(args: ParserArgs) {
+    this.axiosMidgard = axios.create({ baseURL: args.midgardUrl })
+  }
+
+  async parse(tx: Tx): Promise<TxSpecific | undefined> {
+    const opReturn = tx.vout.find(vout => vout.opReturn)?.opReturn
+    if (!opReturn) return
+
+    const regex = /OP_RETURN \(([^)]+)\)/
+
+    const [, memo] = opReturn.match(regex) ?? [undefined, undefined]
+
+    if (!memo) return
+
+    const [type] = memo.split(':')
+
+    switch (type.toLowerCase()) {
```

Comment:
Not sure if that applies to UTXO Txs as well, but that was an LP add liquidity Tx for sym, on the THOR side:

https://viewblock.io/thorchain/tx/A35BD0981D1068903850A752620F9D70B56F3C47A8BD4670D51EE4E3D110F106

Note the `c:` type